# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLFramebuffer.h
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 dom/canvas/WebGLFramebuffer.h
--- a/dom/canvas/WebGLFramebuffer.h	Wed Jan 08 21:52:03 2020 +0000
+++ b/dom/canvas/WebGLFramebuffer.h	Wed Jan 08 22:19:14 2020 +0000
@@ -97,9 +97,8 @@
 
   void DoAttachment(gl::GLContext* gl) const;
 
-  JS::Value GetParameter(WebGLContext* webgl, JSContext* cx, GLenum target,
-                         GLenum attachment, GLenum pname,
-                         ErrorResult* const out_error) const;
+  MaybeWebGLVariant GetParameter(WebGLContext* webgl, GLenum target,
+                                 GLenum attachment, GLenum pname) const;
 
   bool IsEquivalentForFeedback(const WebGLFBAttachPoint& other) const {
     if (!HasAttachment() | !other.HasAttachment()) return false;
@@ -135,8 +134,7 @@
   };
 };
 
-class WebGLFramebuffer final : public nsWrapperCache,
-                               public WebGLRefCountedObject<WebGLFramebuffer>,
+class WebGLFramebuffer final : public WebGLRefCountedObject<WebGLFramebuffer>,
                                public LinkedListElement<WebGLFramebuffer>,
                                public SupportsWeakPtr<WebGLFramebuffer>,
                                public CacheInvalidator {
@@ -194,15 +192,10 @@
   ////
 
  public:
-  NS_INLINE_DECL_CYCLE_COLLECTING_NATIVE_REFCOUNTING(WebGLFramebuffer)
-  NS_DECL_CYCLE_COLLECTION_SCRIPT_HOLDER_NATIVE_CLASS(WebGLFramebuffer)
+  NS_INLINE_DECL_REFCOUNTING(WebGLFramebuffer)
 
   WebGLFramebuffer(WebGLContext* webgl, GLuint fbo);
 
-  WebGLContext* GetParentObject() const { return mContext; }
-  virtual JSObject* WrapObject(JSContext* cx,
-                               JS::Handle<JSObject*> givenProto) override;
-
  private:
   ~WebGLFramebuffer() {
     DeleteOnce();
@@ -272,12 +265,11 @@
   FBStatus CheckFramebufferStatus() const;
   void FramebufferAttach(GLenum attachEnum,
                          const webgl::FbAttachInfo& toAttach);
-  void DrawBuffers(const dom::Sequence<GLenum>& buffers);
+  void DrawBuffers(const nsTArray<GLenum>& buffers);
   void ReadBuffer(GLenum attachPoint);
 
-  JS::Value GetAttachmentParameter(JSContext* cx, GLenum target,
-                                   GLenum attachment, GLenum pname,
-                                   ErrorResult* const out_error);
+  MaybeWebGLVariant GetAttachmentParameter(GLenum target, GLenum attachment,
+                                           GLenum pname);
 
   static void BlitFramebuffer(WebGLContext* webgl, GLint srcX0, GLint srcY0,
                               GLint srcX1, GLint srcY1, GLint dstX0,